wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -136,32 +134,13 @@ class WirePlugin : Plugin<Project> {
 
     addWireRuntimeDependency(hasJavaOutput, hasKotlinOutput)
 
-    val protoPathInput = WireInput(project.configurations.getByName("protoPath"))
-    protoPathInput.addTrees(project, extension.protoTrees)
-    protoPathInput.addJars(project, extension.protoJars)
-    protoPathInput.addPaths(project, extension.protoPaths)
-
     sources.forEach { source ->
-      val protoSourceInput = WireInput(
-        configuration = project.configurations.getByName("protoSource").copy().also {
-          it.isCanBeConsumed = false
-        },
-      )
-      protoSourceInput.addTrees(project, extension.sourceTrees)
-      protoSourceInput.addJars(project, extension.sourceJars)
-      protoSourceInput.addPaths(project, extension.sourcePaths)
       // TODO(Benoit) Should we add our default source folders everytime? Right now, someone could
       //  not combine a custom protoSource with our default using variants.
-      if (protoSourceInput.dependencies.isEmpty()) {
-        protoSourceInput.addPaths(project, defaultSourceFolders(source))
-      }
-
-      val inputFiles = project.layout.files(protoSourceInput.inputFiles, protoPathInput.inputFiles)
-
-      val projectDependencies =
-        (protoSourceInput.dependencies + protoPathInput.dependencies).filterIsInstance<ProjectDependency>()
-      for (projectDependency in projectDependencies) {
-        projectDependenciesJvmConfiguration.dependencies.add(projectDependency)
+      if (extension.protoSourceProtoRootSets.all { it.roots.isEmpty() }) {
```

Comment:
Note: this probably resolves configurations eagerly. I think this is the case already so not a huge deal in the grand scheme of things but would be a good candidate for future improvements.

Reply:
Great idea. We can move this behaviour to execution time.

Reply:
In follow-up!